### DISCUSSION TITLE
UML 클래스 다이어그램 관련 질문

### DISCUSSION THREAD

**Discussion by fkdl0048** [fkdl0048/Mentoring-Project#40] [Q&A] [answered]
UML 클래스 다이어그램관련 내용을 자꾸 접하다보니 해당 내용을 공부해야하는지 조금 헷갈립니다.  

UML 클래스 다이어그램은 필수적으로 알고 가야하는 내용일까요?

**Comment by jongfeel** [comment, marked as answer]
결론부터 말씀드리면, 필수는 아닙니다.
설계하는데 있어서 글로 충분히 표현할 수 있다면 굳이 UML로 먼저 표현하지 않아도 됩니다.
다만, 설계 상에서 도식화된 그림이 필요하다고 판단되면 그 때 사용하는게 좋습니다.

UML에 포함된 다이어그램은 10종류가 넘으므로
내가 무엇을 표현하고 싶은지에 따라 선택적으로 다이어그램을 그려야 합니다.
